wordpress-mobile/WordPress-iOS
# Move persistent store to app group

Moves the persisten store to the App Group so that it can be used by libraries to implement services.

## Testing:

- Please, test all scenarios with the feature flag `.swiftCoreData` ON and OFF.  We need to make sure nothing is broken.
- For all tests I recommend to filter Xcode’s console log by this text “org.wordpress.StoreToAppGroupMigration” 

### Test 1: Fresh install

1. Delete WP from your phone
2. Run the App

Make sure you see this message in the log:

> org.wordpress.StoreToAppGroupMigration: Persistent store does not exist at source location file:///var/mobile/Containers/Data/Application/425CE222-4755-4133-8AE4-F4BBF6662C76/Documents/WordPress.sqlite. We assume this is the first App run.

3. Launch the app again.

Make sure you see this message in the log:

> org.wordpress.StoreToAppGroupMigration: Migration already performed.

### Test 2: Existing install

Do a clean run of the App by deleting it from the device.

1. Check out branch `release/16.0`.  This includes a version of the data model several versions old.
2. Run the app, and login.
3. Kill the App from the device.
4. Check out branch `try/move-persistent-store-to-app-group`.
5. Run the app.

Make sure you see this message in the log.

> org.wordpress.StoreToAppGroupMigration: Persistent store moved successfully to file:///private/var/mobile/Containers/Shared/AppGroup/93E1CD1A-0140-4B75-AC6B-EF138B346F89/WordPress.sqlite

6. Launch the app again.

Make sure you see this message in the log:

> org.wordpress.StoreToAppGroupMigration: Migration already performed.

## PR submission checklist:
